src/llama-model.cpp

```diff
@@ -853,43 +853,16 @@ void llama_model::load_hparams(llama_model_loader & ml) {
                     default: type = LLM_TYPE_UNKNOWN;
                 }
 
-                // for backward compatibility ; see: https://github.com/ggerganov/llama.cpp/pull/8931
-                if ((hparams.n_layer == 32 || hparams.n_layer == 40) && hparams.n_ctx_train == 4096) {
-                    // default value for Phi-3-mini-4k-instruct and Phi-3-medium-4k-instruct
-                    LLAMA_LOG_WARN("%s: assuming n_swa = 2047 for Phi-3-mini-4k-instruct and Phi-3-medium-4k-instruct\n", __func__);
```

Comment:
Sorry I didn't saw your review request.

Hmm I think these configs still need to be kept. The original issue talked about phi-4 and not phi-3, and these branches of code was to make sure old phi-3 model works correctly.

Probably phi-4 conflicts with this but I'll have a look. In anw, I think `n_swa` should still be set, so that the model can mask out tokens correctly for long sequence.

Reply:
The problem is that `n_swa` value never does anything when `n_swa_pattern == 1`, which has always been like this. So unless I am missing something, these configs didn't do anything.

Reply:
Hmm ok it seems like `n_swa_pattern = 1` meaning all layers use SWA. I had a look at [modeling_phi3.py](https://huggingface.co/microsoft/Phi-3-mini-4k-instruct/blob/main/modeling_phi3.py#L459-L463) and seems like `use_sliding_windows` is set to the same value globally.

In anw, I think we can just remove SWA altogether for phi-3/phi-4 if you feel like we spent too much efforts fixing this.